app/src/org/commcare/tasks/DataPullTask.java

```diff
@@ -509,6 +569,32 @@ private void updateUserSyncToken(String syncToken) throws StorageFullException {
         }
     }
 
+    private void parseProgressFromRetryResult(RemoteDataPullResponse response) {
+        try {
+            InputStream stream = response.writeResponseToCache(context).retrieveCache();
+            KXmlParser parser = ElementParser.instantiateParser(stream);
+            parser.next();
+            int eventType = parser.getEventType();
+            do {
+                if (eventType == KXmlParser.START_TAG) {
+                    if (parser.getName().toLowerCase().equals("progress")) {
+                        serverProgressCompletedSoFar = Integer.parseInt(parser.getAttributeValue(null, "done"));
+                        serverProgressTotal = Integer.parseInt(parser.getAttributeValue(null, "total"));
+                        return;
+                    }
+                }
+                eventType = parser.next();
+            } while (eventType != KXmlParser.END_DOCUMENT);
+        } catch (IOException | XmlPullParserException e) {
+            Logger.log(AndroidLogger.TYPE_USER, "Error while parsing progress values of retry result");

```

Comment:
Slightly curious what happens if we get here in terms of the restore process? Does the progress bar enter a specific state? Will the restore continue without updating the progress vars, or will that cause things to abort?